### PR TITLE
Sourcery refactors II : `utils`, `toas`, `toa_select`, `solar_system_ephemerides`, `pulsar_mjd`, and `fermi_toas`

### PR DESCRIPTION
**Pull request by abhisrkckl**
See #1555 

Also fixes a bug in `get_Fermi_TOAs()`. 

EDIT: The bug is already fixed in #1543.

### CODE REVIEW

**Comment by dlakaplan** [comment]
What was the bug?

File: src/pint/fermi_toas.py [hunk 1]
```diff
@@ -379,28 +379,15 @@ def get_Fermi_TOAs(
...
+        t,
+        obs,
+        include_gps=False,
+        include_bipm=False,
+        planets=planets,
+        ephem=ephem,
+        flags=[
+            {"energy": str(e), "weight": str(w)}
+            for e, w in zip(energies.to_value(u.MeV), weights)
+        ],
```
Comment (abhisrkckl) [inline src/pint/fermi_toas.py:392]: @dlakaplan The bug was in this line. "w" was undefined before.

**Comment by abhisrkckl** [comment]
An undefined variable.

**Comment by dlakaplan** [comment]
Seems fine to me.

**Comment by abhisrkckl** [comment]
This can be merged.

**Pull request merged**